### PR TITLE
Refactor-tablesearchcontroller.php-switch cases #11991

### PR DESCRIPTION
**Pull request by flash1452**
Before submitting pull request, please check that every commit:
- [x] Has proper Signed-Off-By
- [x] Has commit message which describes it
- [x] Is needed on it's own, if you have just minor fixes to previous commits, you can squash them

Signed-off-by: Vishal Pandey vishpandey2014@gmail.com

### CODE REVIEW

**Comment by flash1452** [comment]
Signed-Off-By: Vishal Pandey vishpandey2014@gmail.com

**Comment by flash1452** [comment]
@zixtor  Please review the pull request.

File: libraries/controllers/table/TableSearchController.php [hunk 1]
```diff
@@ -169,6 +169,8 @@ private function _loadTableInfo()
      */
     public function indexAction()
     {
+        $temolate_sec_tab=Template::get('secondary_tabs');
```
Comment (ibennetch) [inline libraries/controllers/table/TableSearchController.php]: Is "temolate" a typo here? I suspect you meant "template".

File: libraries/controllers/table/TableSearchController.php [hunk 2]
```diff
@@ -182,18 +184,16 @@ public function indexAction()
...
-                        array(
-                            'url_params' => array(
-                                'db'    => $this->db,
-                                'table' => $this->table,
-                            ),
-                            'sub_tabs'   => $this->_getSubTabs(),
-                        )
+
+
+             $template_sec_tab->set(array(
```
Comment (ibennetch) [inline libraries/controllers/table/TableSearchController.php]: It's tough to see for sure, but I believe your indentation has one extra space here.

File: libraries/controllers/table/TableSearchController.php [hunk 3]
```diff
@@ -182,18 +184,16 @@ public function indexAction()
...
+                        'url_params' => array(
+                            'db'    => $this->db,
+                            'table' => $this->table,
+                        ),
+                        'sub_tabs'   => $this->_getSubTabs(),
                     )
-            );
+             );
```
Comment (ibennetch) [inline libraries/controllers/table/TableSearchController.php]: If there's an extra space indented above, then it appears there's an extra one here, as well.

File: libraries/controllers/table/TableSearchController.php [hunk 4]
```diff
@@ -322,22 +354,17 @@ public function indexAction()
...
-                    ->render(
-                        array(
-                            'url_params' => array(
-                                'db'    => $this->db,
-                                'table' => $this->table,
-                            ),
-                            'sub_tabs'   => $this->_getSubTabs(),
-                        )
+
+                $template_sec_tab->set(array(
```
Comment (ibennetch) [inline libraries/controllers/table/TableSearchController.php]: And I believe this section is indented a level or two too far (two or four spaces too many).

**Comment by ibennetch** [comment]
I've made a few notes, but overall it looks pretty promising to me. Someone else may have some other feedback, of course. I'm not really sure what's going on with the extra merge commits that shouldn't be appearing in the pull request nor am I confident of what the best resolution to that will be.

**Comment by ibennetch** [comment]
I've made a few notes, but overall it looks pretty promising to me. Someone else may have some other feedback, of course.

I'm not really sure what's going on with the extra merge commits that shouldn't be appearing in the pull request nor am I confident of what the best resolution to that will be. It may be that the developer who merges your commit will have to cherry-pick your actual commit, or perhaps your branch history can be fixed outright.

Anyway, the best way to add the sign-off message is to pass `-s` to git commit; such as `git commit -s`. My guess is that you tried to add it manually and using `-s` which lead to the double sign-off message in your commit message. This can be easily edited with `git rebase -i` (or, probably, `git rebase -i -s` if I had to guess) and using "reword" instead of "pick" in the resulting dialog.

**Comment by flash1452** [comment]
@ibennetch Thank You Sir for your suggestions.I have been having some issues with my git commit and squashing.Also I am not sure how this mere commit has appeared ,I don't know but am pretty sure i did not make any merge commits with the master.

**Comment by zixtor** [comment]
It is now showing 775 files changed, please resolve that before further review.

**Pull request closed without merging**